XtRay.Windows/MainWindow.xaml.cs

```diff
@@ -19,12 +21,21 @@ public partial class MainWindow : Window
     {
         private TraceBox traceBox;
         private FlexibleTraceNode rootNode;
+        #region Define parserList
+
+        private ObservableCollection<string> ParserList = new ObservableCollection<string>
+        {
+            "Function", "Parameter", "Return"
+        };
+
+        #endregion
 
         public MainWindow()
         {
             InitializeComponent();
-            var textChanged = ((EventHandler<TextChangedEventArgs>)SearchBox_TextChanged).Debounce(444);
+            var textChanged = ((EventHandler<EventArgs>)ApplyFilterEvent).Debounce(444);
             SearchBox.TextChanged += (s, e) => textChanged(s, e);
+            ComboBox.SelectionChanged += (s, e) => textChanged(s, e);
```

Comment:
Move this line after the line where you set the ItemsSource(`ComboBox.ItemsSource = ParserList;`) and the SelectedIndex, that way the event won't fire programmatically.